vaapi/vaapidisplay.cpp

```diff
@@ -318,13 +308,9 @@ DisplayPtr DisplayCache::createDisplay(const NativeDisplay& nativeDisplay, const
         return vaapiDisplay;
     }
 
-    if (!driverName.empty())
-        vaSelectDriver(vaDisplay, driverName);
+    if (nativeDisplay.type == NATIVE_DISPLAY_VA || vaInit(vaDisplay))
+        vaapiDisplay.reset(new VaapiDisplay(nativeDisplayObj, vaDisplay));
 
-    if (nativeDisplay.type == NATIVE_DISPLAY_VA || vaInit(vaDisplay)) {
```

Comment:
Hi @keqiaox , you should keep this insead line 311 and 312. else klockwork  will complain about something